### PR TITLE
feat: Drawer component

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds `<Drawer />` component

## ✅ Checklist

Easy UI has certain UX standards that must be met. In general, non-trivial changes should meet the following criteria:

- [x] Visuals match Design Specs in Figma
- [x] Stories accompany any component changes
- [x] Code is in accordance with our style guide
- [x] Design tokens are utilized
- [x] Unit tests accompany any component changes
- [x] TSDoc is written for any API surface area
- [x] Specs are up-to-date
- [x] Console is free from warnings
- [x] No accessibility violations are reported
- [x] Cross-browser check is performed (Chrome, Safari, Firefox)
- [x] Changeset is added

~Strikethrough~ any items that are not applicable to this pull request.

### CODE REVIEW

File: easy-ui-react/src/Drawer/Drawer.mdx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+<ArgTypes of={Drawer.BanneredContentArea} />
+
+### Drawer.StandaloneContentArea
+
+<ArgTypes of={Drawer.StandaloneContentArea} />
+
+### Drawer.CloseButton
+
+<ArgTypes of={Drawer.CloseButton} />
```
Comment (kevinalexliu) [inline easy-ui-react/src/Drawer/Drawer.mdx:155]: On the storybook, this doesn't show anything, is this intentional?

File: easy-ui-react/src/Drawer/DrawerCloseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const handleClick = useCallback(() => {
+    drawerTriggerContext.state.close();
+  }, [drawerTriggerContext.state]);
+
+  if (!drawerTriggerContext.isDismissable) {
+    return null;
+  }
+
+  return (
+    <button className={styles.closeButton} onClick={handleClick}>
```
[the remark this answers is not in the document]
Comment (stephenjwatkins) [reply easy-ui-react/src/Drawer/DrawerCloseButton.tsx:21]: the accessible label for the button is taken care of via the `<Text visuallyHidden />` child. with a visually hidden text the `aria-label` isn't necessary.

File: easy-ui-react/src/Drawer/Drawer.mdx (hunk 1 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/Drawer/Drawer.mdx:155]: yeah it has no props. i'll remove it since it's confusing

**Review by kevinalexliu — APPROVED** [review]
LGTM

**Pull request merged**